client/lib/url/omit-url-params.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { URL as URLString } from 'types';
+import { Falsy } from 'utility-types';
+
+/**
+ * Internal dependencies
+ */
+import { determineUrlType, URL_TYPE } from './url-type';
+import format from './format';
+
+const BASE_URL = 'http://__domain__.invalid';
+
+/**
+ * Removes given params from a url.
+ *
+ * @param  url URL to be cleaned
+ * @param  paramsToOmit The collection of params or single param to reject
+ * @return Url less the omitted params.
+ */
+export default function omitUrlParams( url: Falsy, paramsToOmit: string | string[] ): null;
+export default function omitUrlParams( url: URLString, paramsToOmit: string | string[] ): URLString;
+export default function omitUrlParams(
+	url: URLString | Falsy,
+	paramsToOmit: string | string[]
+): URLString | null {
+	if ( ! url ) {
+		// Note that we return null for the valid empty string URL.
+		// This is done to maintain compatibility with the previous implementation.
+		return null;
+	}
+
+	const urlType = determineUrlType( url );
+
+	if ( urlType === URL_TYPE.INVALID ) {
+		return null;
+	}
+
+	let toOmit: string[];
+
+	if ( typeof paramsToOmit === 'string' ) {
+		toOmit = [ paramsToOmit ];
+	} else {
+		toOmit = paramsToOmit;
+	}
+
+	const parsed = new URL( url, BASE_URL );
+	const filtered = Array.from( parsed.searchParams.entries() ).filter(
+		( [ key ] ) => ! toOmit || ! toOmit.includes( key )
```

Comment:
🤔 `! toOmit` doesn't look quite right, `toOmit` is _always_ and array, and `! toOmit` should _always_ evaluate to `false`. We'll always hit the RHS of this `||`:

```suggestion
		( [ key ] ) => ! toOmit.includes( key )
```

Was this intended to be an optimization? I believe we could return the string unmodified after the valid URL check and avoid further processing:

```ts
	if ( urlType === URL_TYPE.INVALID ) {
		return null;
	}

	// An empty string or array of paramsToOmit is a no-op (with a valid URL)
	if ( ! paramsToOmit.length ) {
		return url;
	}
```

(`""` can't remove any params, can it?)

Reply:
The `! toOmit` looks like old code that I forgot to remove after I rewrote the logic here. Thanks!

I also added your suggested early check 👍